docs/k3s.md

```diff
@@ -0,0 +1,15 @@
+At the heart of RKE2 is the embedded K3s engine which functions as a
+supervisor for the kubelet and containerd processes. The K3s engine also
+provides an add-on controller that RKE2 leverages. So, RKE2 depends on K3s,
+but what does that look like from version to version? It is not yet as simple
+as 1.20.7+k3s1rke2r1 &rarr; 1.20.7+k3s1 but starting with 1.22.x it should be.
```

Comment:
Is this right? Or should be this?
```suggestion
as 1.20.7+rke2r1 &rarr; 1.20.7+k3s1 but starting with 1.22.x it should be.
```

Reply:
Since we plan to make this developer docs could we elaborate more? This doesn't yet really explain _why_ the branches are not 1:1 aligned.
I think we should attempt to explain the rationale for the release branches not matching. Keep it simple enough that we don't explain every little reason, but get across the top one or two reasons.

Reply:
A full explanation probably requires covering a bit of the history of the project, and describing the way that enhancements and bug fixes flow from K3s to RKE2. Doesn't need to be more than a paragraph or two but I think it would help with context.